thoth/user_api/api_v1.py

```diff
@@ -995,6 +999,82 @@ def list_buildlogs(page: int = 0):
     return _do_listing(BuildLogsStore, page)
 
 
+def get_python_package_version_metadata(
+    name: str, version: str, index: str, os_name: str, os_version: str, python_version: str
+) -> typing.Tuple[typing.Dict[str, typing.Any], int]:
+    """Obtain metadata for the given package."""
+    name = PackageVersion.normalize_python_package_name(name)
+    version = PackageVersion.normalize_python_package_version(version)
+    os_name = map_os_name(os_name) or os_name  # "or" to keep typing happy
+    os_version = normalize_os_version(os_name, os_version) or os_version  # "or" to keep typing happy
+
+    parameters = locals()
+    from .openapi_server import GRAPH
+
+    solver_documents = GRAPH.get_solver_document_id_all(
+        name, version, index, os_name=os_name, os_version=os_version, python_version=python_version, sort=True
+    )
+    if not solver_documents:
+        return {"parameters": parameters, "error": "No records found for the given request"}, 404
+
+    solver_store = SolverResultsStore()
+    solver_store.connect()
+    solver_document = solver_store.retrieve_document(solver_documents[0])
+    solver_name = "-".join(solver_document["metadata"]["document_id"].split("-")[:4])
+    solver = OpenShift.parse_python_solver_name(solver_name)
+
+    for solver_entry in solver_document["result"]["tree"]:
+        if (
+            solver_entry["package_name"] == name
+            and solver_entry["package_version"] == version
+            and solver_entry["index_url"] == index
+        ):
+            break
+    else:
```

Comment:
Shouldn't this block be moved by a right tab?

Reply:
Nope, it is for-else. If the given entry is not found, the else part is executed.